src/components/Post/RandomButton.tsx

```diff
@@ -0,0 +1,36 @@
+import { useRouter } from 'next/router';
+
+import { randomSlug } from 'helpers/randomSlug';
+
+export const postRandomButtonTestIds = {
+  root: 'posts__random-button',
+};
+
+const RandomButton = () => {
+  const router = useRouter();
+  const onClick = async () => {
+    const slug = await randomSlug();
+
+    if (slug === '') {
+      router.push('/');
+
+      return;
+    }
+
+    router.push(`/posts/${slug}`);
+  };
+
+  return (
+    <div className="absolute top-0 right-0 p-4">
+      <button
+        className="btn btn-circle"
+        onClick={onClick}
+        data-test-id={postRandomButtonTestIds.root}
+      >
```

Comment:
Should we add a state of click here for blocking the next click before completely loading the current request?

Reply:
Added in https://github.com/nimblehq/til-web/pull/33/commits/e0bce78ff53217a7162632559dd59a0d55ec0083